dwave/cloud/client/base.py

```diff
@@ -1462,7 +1462,23 @@ def _handle_problem_status(self, message, future):
                     # An alternative to making this call here would be to pass
                     # self in with the message
                     if future.solver is None:
-                        future.solver = self.get_solver(identity=message['solver'])
+                        # handle problems with v2 solver representation
+                        # (prior to `graph_id` introduction)
+                        # see: https://github.com/dwavesystems/dwave-cloud-client/issues/727
+                        # TODO: remove after a server-side fix
+                        solver = message['solver']
+                        if isinstance(solver, dict):
```

Comment:
Do we want `collections.abc.Mapping`? Probably overly general...

Reply:
Not needed here, as this message always comes from JSON, so types are pretty limited.